sleuth-camel-core/src/test/java/com/playtika/sleuth/camel/SentEventNotifierTest.java

```diff
@@ -33,33 +32,40 @@
 import org.apache.camel.impl.event.ExchangeCompletedEvent;
 import org.apache.camel.impl.event.ExchangeSentEvent;
 import org.apache.camel.spi.CamelEvent;
-import org.junit.After;
-import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
 
 import static com.playtika.sleuth.camel.SentEventNotifier.EXCHANGE_EVENT_SENT_ANNOTATION;
 import static com.playtika.sleuth.camel.SleuthCamelConstants.EXCHANGE_IS_TRACED_BY_BRAVE;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
 import static org.mockito.Mockito.*;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+@TestInstance(PER_CLASS)
 public class SentEventNotifierTest {
 
-    @Mock
+    @Mock()
     private Tracer tracer;
     @Mock
-    private ErrorParser errorParser;
-    @Mock
     private ThreadLocalSpan threadLocalSpan;
-
     @InjectMocks
     private SentEventNotifier sentEventNotifier;
 
-    @After
+    @BeforeEach
+    public void setUp() {
+        sentEventNotifier = new SentEventNotifier(tracer, threadLocalSpan);
```

Comment:
isn't inject mock covers creation of object?

Reply:
Actually it injects them correctly for the first test, however mocks are reset after each test but new mocks are not injected into SentEventNotifier. This object keeps a reference to the first mock objects...

Reply:
that is very weird, all mocks should be reset, so you can use them as new, can you point of example test which is not working for you?

Reply:
Fixed, with a workaround a explained here: https://github.com/mockito/mockito/issues/1131

Reply:
i wonder why it still not fixed since  on Jun 19, 2017 (as that issue dates)

also aren't even cached mocks are reset each test, so may just reuse cached instances (as it was working previously without workaround)

Reply:
It doesn’t seem to be a bug, but API design, as explained in this SO accepted answer: https://stackoverflow.com/questions/20046210/mockito-injectmocks-strange-behaviour-with-final-fields?answertab=active#tab-top

Not sure why it worked before, something may have changed from Junit 4 to Junit 5.

Reply:
I added back `@InjectMocks`